spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/context/ContextRegionProviderAutoConfiguration.java

```diff
@@ -30,6 +31,7 @@
  * @author Agim Emruli
  */
 @Configuration
+@ConditionalOnAwsCloudEnvironment
```

Comment:
Not sure about this one, too.
Actually I have an integration test based on CloudFormation and stack and I do testing without EC2 and even its mocking.